src/main/java/org/myrobotlab/service/ProgramAB.java

```diff
@@ -596,7 +631,10 @@ public void reloadSession(String path, String userName, String botName, Boolean
     if (sessions.containsKey(botName) && sessions.get(botName).containsKey(userName)) {
       // TODO: will garbage collection clean up the bot now ?
       // Or are there other handles to it?
-      sessions.remove(sessions.get(botName).get(userName));
+      HashMap<String, ChatData> session = new HashMap<String, ChatData>();
+      session.put(userName, sessions.get(botName).get(userName));
+      log.info("{} session removed", sessions);
+      sessions.remove(botName, session);

```

Comment:
Two comments:
First something minor: the log statement "session removed" is before it's actually removed : )
Second: I would have to check myself, but did you actually verify this works? My brain keeps telling me the session you're trying to remove would have a different ObjectID and be therefore not the same as the one you just created to remove it. (I might be wrong on this one.)

Reply:
yep log.info need to move down.
Restartsession() was broken, before this fix. because the old session was not deleteted like before. 
because a syntaxerror inside sessions.remove(string,hasmap<>).

Reply:
If you're sure it works now I'm fine.
Thinking of it: Do we have a unit test for this?